### PR TITLE
rename models based on feedback

### PR DESCRIPTION
**Pull request by m-nash**
# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

File: sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/ManagementGroupContainer.cs [hunk 1]
```diff
@@ -315,7 +315,7 @@ public async virtual Task<Response<bool>> CheckIfExistsAsync(string groupId, Man
         /// <param name="cacheControl"> Indicates whether the request should utilize any caches. Populate the header with &apos;no-cache&apos; value to bypass existing caches. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="groupId"/> or <paramref name="createManagementGroupRequest"/> is null. </exception>
-        public virtual Response<ManagementGroup> CreateOrUpdate(string groupId, CreateManagementGroupRequest createManagementGroupRequest, string cacheControl = null, CancellationToken cancellationToken = default)
+        public virtual Response<ManagementGroup> CreateOrUpdate(string groupId, CreateManagementGroupOptions createManagementGroupRequest, string cacheControl = null, CancellationToken cancellationToken = default)
```
Comment (ShivangiReja) [inline sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/ManagementGroupContainer.cs:318]:
```suggestion
        public virtual Response<ManagementGroup> CreateOrUpdate(string groupId, CreateManagementGroupOptions createManagementGroupOptions, string cacheControl = null, CancellationToken cancellationToken = default)
```

File: sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/ManagementGroupContainer.cs [hunk 2]
```diff
@@ -350,7 +350,7 @@ public virtual Response<ManagementGroup> CreateOrUpdate(string groupId, CreateMa
         /// <param name="cacheControl"> Indicates whether the request should utilize any caches. Populate the header with &apos;no-cache&apos; value to bypass existing caches. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="groupId"/> or <paramref name="createManagementGroupRequest"/> is null. </exception>
-        public async virtual Task<Response<ManagementGroup>> CreateOrUpdateAsync(string groupId, CreateManagementGroupRequest createManagementGroupRequest, string cacheControl = null, CancellationToken cancellationToken = default)
+        public async virtual Task<Response<ManagementGroup>> CreateOrUpdateAsync(string groupId, CreateManagementGroupOptions createManagementGroupRequest, string cacheControl = null, CancellationToken cancellationToken = default)
```
Comment (ShivangiReja) [inline sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/ManagementGroupContainer.cs:353]:
```suggestion
        public async virtual Task<Response<ManagementGroup>> CreateOrUpdateAsync(string groupId, CreateManagementGroupOptions createManagementGroupOptions, string cacheControl = null, CancellationToken cancellationToken = default)
```

File: sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/ManagementGroupContainer.cs [hunk 3]
```diff
@@ -385,7 +385,7 @@ public async virtual Task<Response<ManagementGroup>> CreateOrUpdateAsync(string
         /// <param name="cacheControl"> Indicates whether the request should utilize any caches. Populate the header with &apos;no-cache&apos; value to bypass existing caches. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="groupId"/> or <paramref name="createManagementGroupRequest"/> is null. </exception>
-        public virtual ManagementGroupsCreateOrUpdateOperation StartCreateOrUpdate(string groupId, CreateManagementGroupRequest createManagementGroupRequest, string cacheControl = null, CancellationToken cancellationToken = default)
+        public virtual ManagementGroupsCreateOrUpdateOperation StartCreateOrUpdate(string groupId, CreateManagementGroupOptions createManagementGroupRequest, string cacheControl = null, CancellationToken cancellationToken = default)
```
Comment (ShivangiReja) [inline sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/ManagementGroupContainer.cs:388]:
```suggestion
        public virtual ManagementGroupsCreateOrUpdateOperation StartCreateOrUpdate(string groupId, CreateManagementGroupOptions createManagementGroupOptions, string cacheControl = null, CancellationToken cancellationToken = default)
```

File: sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/ManagementGroupContainer.cs [hunk 4]
```diff
@@ -420,7 +420,7 @@ public virtual ManagementGroupsCreateOrUpdateOperation StartCreateOrUpdate(strin
         /// <param name="cacheControl"> Indicates whether the request should utilize any caches. Populate the header with &apos;no-cache&apos; value to bypass existing caches. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="groupId"/> or <paramref name="createManagementGroupRequest"/> is null. </exception>
-        public async virtual Task<ManagementGroupsCreateOrUpdateOperation> StartCreateOrUpdateAsync(string groupId, CreateManagementGroupRequest createManagementGroupRequest, string cacheControl = null, CancellationToken cancellationToken = default)
+        public async virtual Task<ManagementGroupsCreateOrUpdateOperation> StartCreateOrUpdateAsync(string groupId, CreateManagementGroupOptions createManagementGroupRequest, string cacheControl = null, CancellationToken cancellationToken = default)
```
Comment (ShivangiReja) [inline sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/ManagementGroupContainer.cs:423]:
```suggestion
        public async virtual Task<ManagementGroupsCreateOrUpdateOperation> StartCreateOrUpdateAsync(string groupId, CreateManagementGroupOptions createManagementGroupOptions, string cacheControl = null, CancellationToken cancellationToken = default)
```

File: sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/ManagementGroupContainer.cs [hunk 5]
```diff
@@ -448,7 +448,7 @@ public async virtual Task<ManagementGroupsCreateOrUpdateOperation> StartCreateOr
         /// <summary> Checks if the specified management group name is valid and unique. </summary>
         /// <param name="checkNameAvailabilityRequest"> Management group name availability check parameters. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public virtual Response<CheckNameAvailabilityResult> CheckNameAvailability(CheckNameAvailabilityRequest checkNameAvailabilityRequest, CancellationToken cancellationToken = default)
+        public virtual Response<CheckNameAvailabilityResult> CheckNameAvailability(CheckNameAvailabilityOptions checkNameAvailabilityRequest, CancellationToken cancellationToken = default)
```
Comment (ShivangiReja) [inline sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/ManagementGroupContainer.cs:451]:
```suggestion
        public virtual Response<CheckNameAvailabilityResult> CheckNameAvailability(CheckNameAvailabilityOptions checkNameAvailabilityOptions, CancellationToken cancellationToken = default)
```

File: sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/ManagementGroupContainer.cs [hunk 6]
```diff
@@ -466,7 +466,7 @@ public virtual Response<CheckNameAvailabilityResult> CheckNameAvailability(Check
         /// <summary> Checks if the specified management group name is valid and unique. </summary>
         /// <param name="checkNameAvailabilityRequest"> Management group name availability check parameters. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
-        public async virtual Task<Response<CheckNameAvailabilityResult>> CheckNameAvailabilityAsync(CheckNameAvailabilityRequest checkNameAvailabilityRequest, CancellationToken cancellationToken = default)
+        public async virtual Task<Response<CheckNameAvailabilityResult>> CheckNameAvailabilityAsync(CheckNameAvailabilityOptions checkNameAvailabilityRequest, CancellationToken cancellationToken = default)
```
Comment (ShivangiReja) [inline sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/ManagementGroupContainer.cs:469]:
```suggestion
        public async virtual Task<Response<CheckNameAvailabilityResult>> CheckNameAvailabilityAsync(CheckNameAvailabilityOptions checkNameAvailabilityOptions, CancellationToken cancellationToken = default)
```

File: sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/RestOperations/ManagementGroupsRestOperations.cs [hunk 7]
```diff
@@ -221,7 +221,7 @@ public Response<ManagementGroupData> Get(string groupId, ManagementGroupExpandTy
             }
         }
 
-        internal HttpMessage CreateCreateOrUpdateRequest(string groupId, CreateManagementGroupRequest createManagementGroupRequest, string cacheControl)
+        internal HttpMessage CreateCreateOrUpdateRequest(string groupId, CreateManagementGroupOptions createManagementGroupRequest, string cacheControl)
```
Comment (ShivangiReja) [inline sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/RestOperations/ManagementGroupsRestOperations.cs:224]:
```suggestion
        internal HttpMessage CreateCreateOrUpdateRequest(string groupId, CreateManagementGroupOptions createManagementGroupOptions, string cacheControl)
```

File: sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/RestOperations/ManagementGroupsRestOperations.cs [hunk 8]
```diff
@@ -254,7 +254,7 @@ internal HttpMessage CreateCreateOrUpdateRequest(string groupId, CreateManagemen
         /// <param name="cacheControl"> Indicates whether the request should utilize any caches. Populate the header with &apos;no-cache&apos; value to bypass existing caches. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="groupId"/> or <paramref name="createManagementGroupRequest"/> is null. </exception>
-        public async Task<ResponseWithHeaders<ManagementGroupsCreateOrUpdateHeaders>> CreateOrUpdateAsync(string groupId, CreateManagementGroupRequest createManagementGroupRequest, string cacheControl = null, CancellationToken cancellationToken = default)
+        public async Task<ResponseWithHeaders<ManagementGroupsCreateOrUpdateHeaders>> CreateOrUpdateAsync(string groupId, CreateManagementGroupOptions createManagementGroupRequest, string cacheControl = null, CancellationToken cancellationToken = default)
```
Comment (ShivangiReja) [inline sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/RestOperations/ManagementGroupsRestOperations.cs:257]:
```suggestion
        public async Task<ResponseWithHeaders<ManagementGroupsCreateOrUpdateHeaders>> CreateOrUpdateAsync(string groupId, CreateManagementGroupOptions createManagementGroupOptions, string cacheControl = null, CancellationToken cancellationToken = default)
```

File: sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/RestOperations/ManagementGroupsRestOperations.cs [hunk 9]
```diff
@@ -288,7 +288,7 @@ public async Task<ResponseWithHeaders<ManagementGroupsCreateOrUpdateHeaders>> Cr
         /// <param name="cacheControl"> Indicates whether the request should utilize any caches. Populate the header with &apos;no-cache&apos; value to bypass existing caches. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentNullException"> <paramref name="groupId"/> or <paramref name="createManagementGroupRequest"/> is null. </exception>
-        public ResponseWithHeaders<ManagementGroupsCreateOrUpdateHeaders> CreateOrUpdate(string groupId, CreateManagementGroupRequest createManagementGroupRequest, string cacheControl = null, CancellationToken cancellationToken = default)
+        public ResponseWithHeaders<ManagementGroupsCreateOrUpdateHeaders> CreateOrUpdate(string groupId, CreateManagementGroupOptions createManagementGroupRequest, string cacheControl = null, CancellationToken cancellationToken = default)
```
Comment (ShivangiReja) [inline sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/RestOperations/ManagementGroupsRestOperations.cs:291]:
```suggestion
        public ResponseWithHeaders<ManagementGroupsCreateOrUpdateHeaders> CreateOrUpdate(string groupId, CreateManagementGroupOptions createManagementGroupOptions, string cacheControl = null, CancellationToken cancellationToken = default)
```

File: sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/RestOperations/ManagementGroupsRestOperations.cs [hunk 10]
```diff
@@ -764,7 +764,7 @@ public Response<DescendantListResult> GetDescendantsNextPage(string nextLink, st
             }
         }
 
-        internal HttpMessage CreateCheckNameAvailabilityRequest(CheckNameAvailabilityRequest checkNameAvailabilityRequest)
+        internal HttpMessage CreateCheckNameAvailabilityRequest(CheckNameAvailabilityOptions checkNameAvailabilityRequest)
```
Comment (ShivangiReja) [inline sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/RestOperations/ManagementGroupsRestOperations.cs:767]:
```suggestion
        internal HttpMessage CreateCheckNameAvailabilityRequest(CheckNameAvailabilityOptions checkNameAvailabilityOptions)
```

**Review by allenjzhang — APPROVED** [review]
With Shivangi's comment.

**Review by ShivangiReja — APPROVED** [review]

File: sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/RestOperations/ManagementGroupsRestOperations.cs (hunk 7 above)
Comment (ShivangiReja) [reply sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/RestOperations/ManagementGroupsRestOperations.cs]: You forgot to change parameter names in this class :)

File: sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/RestOperations/ManagementGroupsRestOperations.cs (hunk 10 above)
Comment (ShivangiReja) [reply sdk/resourcemanager/Azure.ResourceManager/src/Generated/ManagementGroup/RestOperations/ManagementGroupsRestOperations.cs:767]: You forgot to update this one :)

**Pull request merged**